postreise/plot/plot_bar_generation_max_min_actual.py

```diff
@@ -32,10 +34,37 @@ def plot_bar_generation_max_min_actual(
     :param int/float fontsize: font size of the texts shown on the plot.
     :param plot_show: show the plot or not, defaults to True.
     :return: (*matplotlib.axes.Axes*) -- axes object of the plot.
+    :raises TypeError:
+        if ``interconnect`` and ``gen_type`` are not str
+        if ``fontsize`` is not an integer or a float and/or
+    :raises ValueError: if ``interconnect is not valid
     """
-    grid = scenario.state.get_grid()
+    _check_scenario_is_in_analyze_state(scenario)
+    if not isinstance(interconnect, str):
+        raise TypeError("interconnect must be a str")
+    if not isinstance(gen_type, str):
+        raise TypeError("gen_type must be a str")
+    if not isinstance(fontsize, (int, float)):
+        raise TypeError("fontsize must be either a int or a float")
+
+    grid = scenario.get_grid()
+    _check_resources_are_in_grid_and_format(gen_type, grid)
+
     plant = grid.plant[grid.plant.type == gen_type]
     mi = ModelImmutables(scenario.info["grid_model"])
+    if interconnect not in mi.zones["interconnect"]:
```

Comment:
Could we do `mi = grid.model_immutables` here?

Reply:
Indeed

Reply:
Done